lib/SILGen/SILGenConstructor.cpp

```diff
@@ -1204,9 +1204,45 @@ void SILGenFunction::emitClassConstructorInitializer(ConstructorDecl *ctor) {
     }
   }
 
+  CleanupStateRestorationScope SelfCleanupSave(Cleanups);
+
+  // If we're using a box for self, reload the value at the end of the init
+  // method.
+  auto cleanupLoc = CleanupLocation(ctor);
+  if (NeedsBoxForSelf) {
+    ManagedValue storedSelf =
+        ManagedValue::forBorrowedAddressRValue(VarLocs[selfDecl].value);
+    selfArg = B.createLoadCopy(cleanupLoc, storedSelf);
+  } else {
+    // We have to do a retain because we are returning the pointer +1.
+    //
+    // SEMANTIC ARC TODO: When the verifier is complete, we will need to
+    // change this to selfArg = B.emitCopyValueOperation(...). Currently due
+    // to the way that SILGen performs folding of copy_value, destroy_value,
+    // the returned selfArg may be deleted causing us to have a
+    // dead-pointer. Instead just use the old self value since we have a
+    // class.
+    selfArg = B.createCopyValue(cleanupLoc, selfArg);
+  }
+
   // Some distributed actor initializers need to init the actorSystem & id now
   if (isDesignatedDistActorInit) {
-    emitDistributedActorImplicitPropertyInits(ctor, selfArg);
+    // If we're using a box for self, reload the value at the end of the init
+    // method.
+    ManagedValue selfValue = selfArg;
+//    if (NeedsBoxForSelf) {
+//      auto storedSelf =
+//          ManagedValue::forBorrowedAddressRValue(VarLocs[selfDecl].value);
+//      selfValue =
+//          B.createLoadCopy(VarLocs[selfDecl].value.getLoc(), storedSelf);
+//      fprintf(stderr, "[%s:%d](%s) the copy \n", __FILE_NAME__, __LINE__, __FUNCTION__);
+//      selfValue.dump();
+//      SelfCleanupSave.pushCleanupState(selfValue.getCleanup(),
+//                                       CleanupState::Dormant);
+//    } else {
+//      selfValue = selfArg;
+//    }
```

Comment:
Alternative naive attempt, just trying it out -- would have been to copy here, but the cleanups are not right then 🤔